pkg/virt-controller/services/template.go

```diff
@@ -62,6 +70,66 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VM) (*kubev1.Pod, error) {
 	return &pod, nil
 }
 
+func (t *templateService) RenderMigrationJob(vm *v1.VM, sourceNode *kubev1.Node, targetNode *kubev1.Node) (*batch.Job, error) {
+	found := false
+	srcAddr := ""
+	dstAddr := ""
+	for _, addr := range sourceNode.Status.Addresses {
+		if addr.Type == kubev1.NodeExternalIP {
+			srcAddr = addr.Address
+			found = true
+		}
+	}
+	if !found {
+		err := fmt.Errorf("migration source node is unreachable")
+		logging.DefaultLogger().Error().Reason(err).Msg("source node does not have external ip")
+		return nil, err
+	}
+	srcUri := fmt.Sprintf("qemu+tcp://%s", srcAddr)
+
+	found = false
+	for _, addr := range targetNode.Status.Addresses {
+		if addr.Type == kubev1.NodeExternalIP {
```

Comment:
I think `kubev1.NodeHostName` is the better choice here, since we have dns. Let's try that first. If that does not work for us, we should probably go with `kubev1.NodeInternalIP` since we migrate from within the cluster.